includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -1466,6 +1493,65 @@ class="jetpack-logo"
 		<?php
 	}
 
+	protected function get_all_activate_errors() {
+		return array(
+			'default' => __( "Sorry! We tried, but we couldn't connect Jetpack just now 😭. Please go to the Plugins tab to connect Jetpack, so that you can finish setting up your store.", 'woocommerce' ),
+			'jetpack_cant_be_installed' => __( "Sorry! We tried, but we couldn't install Jetpack for you 😭. Please go to the Plugins tab to install it, and finish setting up your store.", 'woocommerce' ),
+			'jetpack_cant_be_activated' => __( "Sorry! We tried, but we couldn't activate Jetpack for you 😭. Please go to the Plugins tab to activate it, and finish setting up your store.", 'woocommerce' ),
+			'register_http_request_failed' => __("Sorry! We couldn't contact Jetpack just now 😭. Please make sure that your site is visible over the internet, and that it accepts incoming and outgoing requests via curl. You can also try to connect to Jetpack again, and if you run into any more issues, please contact support.", 'woocommerce'),
+			'siteurl_private_ip_dev' => __( "Whoops! We couldn't connect. Your site is probably on a private network. Jetpack can only connect to public sites. Please make sure your site is visible over the internet, and then try connecting again 🙏." , 'woocommerce'),
+		);
+	}
+
+	protected function get_activate_error_message( $code = '' ) {
+		$errors = $this->get_all_activate_errors();
+		return array_key_exists( $code, $errors ) ? $errors[ $code ] : $errors['default'];
+	}
+
+	protected function install_and_activate_jetpack_sync() {
+		include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
```

Comment:
Can this use [`WC_Install:: background_installer()`](https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-install.php#L961) instead?

Reply:
No, because then I wouldn't be able to use my verbose error messages specifically for the Jetpack connection. I don't want to go refactoring `WC_Install:: background_installer()` to handle that right now, since it installs much more than the Jetpack plugin.

Reply:
I also added a check for `current_user_can( 'install_plugins' )`.

Reply:
Can you elaborate on how this would effect the Jetpack connection error messaging? This is just for install/activate right?

While `WC_Install:: background_installer()` doesn't return anything meaningful, you could check the status of Jetpack after calling it. That should introduce a lot less code - what do you think?

Reply:
I didn't realize that `WC_Install:: background_installer()` was synchronous now, this turns out to be much easier than I thought before.

Changed to `WC_Install:: background_installer()` in 09ff2b5767d4b775c1842407c41295260eb21e30